pkg/controller/elasticsearch/validation/validations.go

```diff
@@ -42,6 +43,22 @@ func validName(ctx Context) validation.Result {
 	return validation.OK
 }
 
+func specUpdatedToBeta(ctx Context) validation.Result {
+	oldAPIVersion := "elasticsearch.k8s.elastic.co/v1alpha1"
+
+	es := ctx.Proposed.Elasticsearch
+	if es.APIVersion == oldAPIVersion || es.Spec.NodeSets == nil {
+		return validation.Result{Reason: validationFailedMsg}
+	}
+	for _, set := range es.Spec.NodeSets {
+		if set.Count == 0 {
+			return validation.Result{Reason: validationFailedMsg}
```

Comment:
Same here: I vote to include the exact reason for failure. Perhaps something along the lines of:

```go
return validation.Result{Reason: fmt.Sprintf("%s: node count of node set '%s' should not be zero", validationFailedMsg, set.Name)}
```